assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -8,6 +8,7 @@ import {
 	usePaymentMethodDataContext,
 } from '@woocommerce/base-context';
 import RadioControl from '@woocommerce/base-components/radio-control';
+import { PAYMENT_GATEWAY_SORT_ORDER } from '@woocommerce/block-settings';
```

Comment:
`SavedPaymentMethodOptions` is already consuming the Payment Method data context via `usePaymentMethodDataContext`. So you can grab the `paymentMethods` and `paymentMethodsInitialized` properties from there and use those to match against whether saved payment methods should display or not.

I even wonder if this should be handled upstream in the payment method data context and filter the available `customerPaymentMethod` by what is actually enabled in the `paymentMethods` object (which could be done via a effect in the context provider). That way this logic wouldn't need to exist in this component.

`PAYMENT_GATEWAY_SORT_ORDER` creates a coupling to server provided values that I think we should try to limit as much as possible in our components so down the road if this data is provided over REST or a transport other than via a WordPress host there's less refactoring involved.

Reply:
:100: good idea. Should be fixed now.